dts/bindings/mtd/zephyr,i2c-target-eeprom.yaml

```diff
@@ -5,3 +5,11 @@ description: Zephyr I2C target EEPROM
 compatible: "zephyr,i2c-target-eeprom"
 
 include: ["eeprom-base.yaml", i2c-device.yaml]
+
+properties:
+  address-width:
+    type: int
+    enum: [8, 16]
+    description: |
+      Number of address bits used to address the EEPROM. If not specified
+      the EEPROM is assumed to have a 8-bit address.
```

Comment:
We have both size (from eeprom-base.yaml) and address-width, and they should (normally) be in sync. How about figuring out the number of address bytes from the size (E.G. 8bit for <= 256, else 16bit)?

Reply:
@henrikbrixandersen what would be the preferred approach?
I agree with @jacmet it would be "cleaner" if the address width is linked to the actual size. However, making it dynamic means it would break existing use where the size is set larger than 256. Even though right now, it's not possible to address larger sizes. 

I saw that some overlays uses in the eeprom target tests actually have the size set to 1024. These overlays would then also need to be updated.

Reply:
`address-width` and `size` are not necessarily tightly coupled in "real" AT24 EEPROMs. It depends on the family of the EEPROM. 

I don't think we should tightly-couple them here either.

Reply:
> you mean things like 24c04 where the upper address bits "spill" into the I2C address? Indeed, but the driver does not support that today (and some I2C drivers only support 1 or very few target addresses).

Correct, but if we introduce the automatic derivation of one from the other, it will make it more difficult to add support for something like the above later on.

A use-case could be to use Zephyr for emulating (part of) a "real" EEPROM, where the user is not in control of the I2C "master" and thus needs to mimic the address-width of the device, but perhaps not the size. This can be needed due to obsolescence of the original part, redesigns, ...

> Even if not tightly coupled, they are certainly related - E.G. it doesn't make any sense to use a size that cannot be represented in the address byte(s)

As long as we error out if attempting to write past the actual buffer size, I don't see a need for this restriction.

Reply:
Okay, then we can leave it as implemented where size and address-width are not directly linked to each other.

Reply:
OK, fine by me. The current behavior is that the address provided over I2C is used mod size, so it wraps around if "too many" I2C address bits are available. Do you still want to allow bigger buffer sizes than what can be addressed over I2C, E.G. wasted inaccessible memory like the current tests with 1 address byte and 1024 bytes?

Reply:
I'd suggest adding a build assert checking for too big buffers compared to `address-width`.

Reply:
@henrikbrixandersen  OK. That presumably alsa means updating the board overlays in the tests/drivers/i2c/i2c_target_api test that do not set address-width (E.G. = 1 byte) and use a 1024 byte buffer today to use a 256 byte butter?

Reply:
Yes, please. Otherwise they would fail this build assertion.

Reply:
Ok! I'll add the build define. W.r.t the tests, I'll move some tests to size 256 and from some I'll add address-width 16 in order to test both case.

Reply:
Added build-time asserts and fixed the sizes to 256 in the tests.